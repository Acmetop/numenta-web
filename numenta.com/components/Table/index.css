.border {
  composes: br2 from 'tachyons';
  composes: collapse from 'tachyons';
}

.table {
  composes: mb4 from 'tachyons';
}
