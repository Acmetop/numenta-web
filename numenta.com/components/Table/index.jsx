import React from 'react'

import styles from './index.css'


const Table = ({children}) => (
  <table className={styles.table}>
    {children}
  </table>
)

Table.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default Table
