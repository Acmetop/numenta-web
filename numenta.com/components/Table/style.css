.table {
  composes: b--black-10 from 'tachyons';
  composes: bb from 'tachyons';
  composes: br from 'tachyons';
  composes: br2 from 'tachyons';
  composes: collapse from 'tachyons';
  composes: mb4 from 'tachyons';
}
