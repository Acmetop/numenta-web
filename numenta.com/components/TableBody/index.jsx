import React from 'react'

import styles from './style.css'


const TableBody = ({children}) => (
  <tbody className={styles.tableBody}>
    {children}
  </tbody>
)

TableBody.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default TableBody
