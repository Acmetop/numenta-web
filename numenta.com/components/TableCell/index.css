/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableCell {
  composes: lh-title from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: v-top from 'tachyons';
}
