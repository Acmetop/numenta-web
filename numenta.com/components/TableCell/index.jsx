// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const TableCell = ({border, children, colSpan}) => {
  const classes = [styles.tableCell]

  if (border) classes.push(styles.border)

  return (
    <td className={classNames(...classes)} colSpan={colSpan}>
      {children}
    </td>
  )
}

TableCell.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.any,
  colSpan: React.PropTypes.number.isRequired,
}

TableCell.defaultProps = {
  colSpan: 1,
}

export default TableCell
