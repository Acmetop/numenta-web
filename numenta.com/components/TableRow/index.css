.tableRow {
  composes: db from 'tachyons';
  composes: dtc-ns from 'tachyons';
  composes: w-100 from 'tachyons';
  composes: w-auto-ns from 'tachyons';
}
