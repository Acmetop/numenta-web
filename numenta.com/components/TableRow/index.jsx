import React from 'react'

import styles from './style.css'


const TableRow = ({children}) => (
  <tr className={styles.tableRow}>
    {children}
  </tr>
)

TableRow.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default TableRow
