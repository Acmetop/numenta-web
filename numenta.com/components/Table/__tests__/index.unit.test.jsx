import React from 'react'
import renderer from 'react-test-renderer'

import Table from '../../Table'
import TableBody from '../../TableBody'
import TableCell from '../../TableCell'
import TableRow from '../../TableRow'


describe('<Table /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Table>
        <TableBody>
          <TableRow>
            <TableCell>Cell 1</TableCell>
            <TableCell>Cell 2</TableCell>
            <TableCell>Cell 3</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
