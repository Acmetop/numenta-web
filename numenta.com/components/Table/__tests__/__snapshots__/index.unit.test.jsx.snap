exports[`Table React component Renders correctly 1`] = `
<table
  className="table">
  <tbody
    className="tableBody">
    <tr
      className="tableRow">
      <td
        className="tableCell"
        colSpan={1}>
        Cell 1
      </td>
      <td
        className="tableCell"
        colSpan={1}>
        Cell 2
      </td>
      <td
        className="tableCell"
        colSpan={1}>
        Cell 3
      </td>
    </tr>
  </tbody>
</table>
`;
